test/controllers/products_controller_test.rb

```diff
@@ -54,7 +56,7 @@
     end
   end
 
-  describe "create product" do
+  describe "create product with signed in merchant" do
     before do
       @merchant = merchants(:sappy1)
       login(@merchant, :github)
```

Comment:
We can test that the merchant successfully logged in with: flash[:success].must_equal "sappy1 is logged in!"
(it appears also in a part you didn't write on line 42)